tools/binlog/tidb-binlog-kafka.md

```diff
@@ -0,0 +1,421 @@
+---
+title: TiDB-Binlog kafka 部署方案
+category: tools
+---
+
+# TiDB-Binlog 部署方案
+
+本文档介绍如何部署 Kafka 版本的 TiDB-Binlog。
+
+## TiDB-Binlog 简介
+
+TiDB-Binlog 用于收集 TiDB 的 Binlog，并提供实时备份和同步功能的商业工具。
+
+TiDB-Binlog 支持以下功能场景:
+
+* **数据同步**：同步 TiDB 集群数据到其他数据库
+* **实时备份和恢复**：备份 TiDB 集群数据，同时可以用于 TiDB 集群故障时恢复
+
+## TiDB-Binlog 架构
+
+首先介绍 TiDB-Binlog 的整体架构。
+
+![TiDB-Binlog 架构](../../media/tidb_binlog_kafka_architecture.png)
+
+TiDB-Binlog 集群主要分为三个组件：
+
+#### Pump
+
+Pump 是一个守护进程，在每个 TiDB 主机的后台运行。其主要功能是实时记录 TiDB 产生的 Binlog 并顺序写入 Kafka 中。
+
+#### Drainer
+
+Drainer 从 Kafka 中收集 Binlog，并按照 TiDB 中事务的提交顺序转化为指定数据库兼容的 SQL 语句，最后同步到目的数据库或者写到顺序文件。
+
+#### Kafka & ZooKeeper
+
+Kafka 集群用来存储由 Pump 写入的 Binlog 数据，并提供给 Drainer 进行读取。
+
+> **注**：local 版本将 Binlog 存储在文件中，最新版本则使用 Kafka 存储。
+
+## TiDB-Binlog 安装
+
+以下为 TiDB-Ansible 分支与 TiDB 版本的对应关系，版本选择可咨询官方 info@pingcap.com。
+
+    | TiDB-Ansible 分支 | TiDB 版本 | 备注 |
+    | ---------------- | --------- | --- |
+    | release-2.0 | 2.0 版本 | 最新 2.0 稳定版本，可用于生产环境。 |
+
+### 下载官方 Binary
+
+CentOS 7+
+
+```bash
+# 下载压缩包
+wget http://download.pingcap.org/tidb-binlog-kafka-linux-amd64.tar.gz
+wget http://download.pingcap.org/tidb-binlog-kafka-linux-amd64.sha256
+
+# 检查文件完整性，返回 ok 则正确
+sha256sum -c tidb-binlog-kafka-linux-amd64.sha256
+
+# 解开压缩包
+tar -xzf tidb-binlog-kafka-linux-amd64.tar.gz
+cd tidb-binlog-kafka-linux-amd64
+```
+
+### TiDB-Binlog 部署
+
+#### 注意
+
+* 需要为一个 TiDB 集群中的每台 TiDB server 部署一个 Pump，目前 TiDB server 只支持以 unix socket 的方式输出 Binlog。
+
+* 手动部署时，启动优先级为：Pump > TiDB；停止优先级为 TiDB > Pump。
+
+    设置 TiDB 启动参数 `binlog-socket` 为对应的 Pump 参数 `socket` 所指定的 unix socket 文件路径，最终部署结构如下图所示：
+
+    ![TiDB pump 模块部署结构](../../media/tidb-pump-deployment.png)
+
+* Drainer 不支持对 ignore schemas（在过滤列表中的 schemas）的 table 进行 rename DDL 操作。
+
+* 在已有的 TiDB 集群中启动 Drainer，一般需要全量备份并且获取 savepoint，然后导入全量备份，最后启动 Drainer 从 savepoint 开始同步；
+
+    为了保证数据的完整性，在 Pump 运行 10 分钟左右后按顺序进行如下操作：
+
+    *  使用 [tidb-tools](https://github.com/pingcap/tidb-tools) 项目中的 [binlogctl](https://github.com/pingcap/tidb-tools/tree/master/tidb-binlog/binlogctl) 工具生成 Drainer 初次启动所需的 position
+    *  全量备份，例如 mydumper 备份 TiDB
+    *  全量导入备份到目标系统
+    *  Kafka 版本 Drainer 启动的 savepoint 默认保存在下游 database tidb_binlog 下的 checkpoint 表中，如果 checkpoint 表中没有效的数据，可以通过设置 `initial-commit-ts` 启动 Drainer 从指定位置开始消费 - `bin/drainer --config=conf/drainer.toml --initial-commit-ts=${position}`
+
+* Drainer 输出为 pb，要在配置文件中设置如下参数：
+
+    ```toml
+    [syncer]
+    db-type = "pb"
+    disable-dispatch = true
+
+    [syncer.to]
+    dir = "/path/pb-dir"
+    ```
+
+* Drainer 输出为 kafka，要在配置文件中设置如下参数：
+
+    ```toml
+    [syncer]
+    db-type = "kafka"
+
+    # when db-type is kafka, you can uncomment this to config the down stream kafka, or it will be the same kafka addrs where drainer pull binlog from.
+    #[syncer.to]
```

Comment:
should we uncomment it?

Reply:
uncomment these configures will be better since they will be used in kafka version. 
Updated in https://github.com/pingcap/docs-cn/pull/1264/commits/bc86fbe735842348ba57b88568bd887e27c733d1